### PR TITLE
pm: don't (un)block IDLE mode

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

This is already unblocked and will trigger an assertion.
The code is still broken as other modes might be unblocked too, but at least it is just as broken as it was before #17895

### Testing procedure

`examples/lorawan` should no longer crash on most boards.
(should still crash on samd21 boards as it would since ed9f740edde87fb69ed2a214ed6348024088fa2f)

### Issues/PRs references

follow-up to #17895

### CODE REVIEW

File: examples/lorawan/main.c [hunk 1]
```diff
@@ -154,7 +154,7 @@ int main(void)
      * in general provides RAM retention after wake-up.
      */
 #if IS_USED(MODULE_PM_LAYERED)
-    for (unsigned i = 1; i < PM_NUM_MODES; ++i) {
+    for (unsigned i = 1; i < PM_NUM_MODES - 1; ++i) {
```
Comment (kaspar030) [inline examples/lorawan/main.c:157]: so, idle mode was special, then it was not, now it is again. :/

**Review by OlegHahm — APPROVED** [review]
Proxy ACK from @kaspar030  (via Matrix)

**Comment by jia200x** [comment]
let's see if this fixes the LoRaWAN issue

File: pkg/openwsn/contrib/board.c [hunk 2]
```diff
@@ -49,8 +49,8 @@ void board_init_openwsn(void)
     LOG_DEBUG("[openwsn/board]: init\n");
 
 #ifdef MODULE_PM_LAYERED
-    /* sleeping is currently not supported, block all sleep modes */
-    for (uint8_t i = 0; i < PM_NUM_MODES; i++) {
+    /* sleeping is currently not supported, block all sleep modes except IDLE */
+    for (uint8_t i = 0; i < PM_NUM_MODES - 1; i++) {
         pm_block(i);
```
Comment (benpicco) [inline pkg/openwsn/contrib/board.c:54]: tbh I don't see what's the point of this code - sleep modes are blocked by default on RIOT anyway.

**Pull request merged**